ReactAndroid/src/main/java/com/facebook/react/modules/network/RequestBodyUtil.java

```diff
@@ -52,17 +56,32 @@ public static boolean isGzipEncoding(@Nullable final String encodingType) {
    */
   public static @Nullable InputStream getFileInputStream(
       Context context, String fileContentUriStr) {
+        Uri dataContentUri=null;
     try {
       Uri fileContentUri = Uri.parse(fileContentUriStr);
 
       if (fileContentUri.getScheme().startsWith("http")) {
         return getDownloadFileInputStream(context, fileContentUri);
       }
+
+      if (fileContentUriStr.startsWith("data:")) {
+        byte[] decodedDataUrString = Base64.decode(fileContentUriStr.split(",")[1], Base64.DEFAULT);
+        Bitmap bitMap = BitmapFactory.decodeByteArray(decodedDataUrString, 0, decodedDataUrString.length);
+        ByteArrayOutputStream bytes = new ByteArrayOutputStream();
+        bitMap.compress(Bitmap.CompressFormat.JPEG, 100, bytes);  
+        String path = MediaStore.Images.Media.insertImage(context.getContentResolver(), bitMap, "Title", null);
```

Comment:
I tried to look up for more options, and datauri seems not supported out of the box. I agree to create a temp file and use the input stream like so is the correct solution. However, I suggest:

1. Evaluate any impact on large image files
2. Checkout the next method in this file, [getDownloadFileInputStream](https://github.com/facebook/react-native/blob/3199c5801ca41476e4756dbcd00a154dd87b50a8/ReactAndroid/src/main/java/com/facebook/react/modules/network/RequestBodyUtil.java#L91), where it also created a temp file and return the inputstream. Can we somehow re-use the implementationt here?

Reply:
@ryancat  Thanks for the review! I've now converted the dataUri in bitmap and then attached an inputStream to the compressed bitmap. This way we won't have to store the image temporarily. I think converting the dataUri to Bitmap is necessary as there is no method that lets us convert the dataUri to inputStream directly. So regarding large size images, converting them to bitmap is the only(efficient) step we are left with, and the conversion is handled by in-built java function